lib/phases/cold-launch.js

```diff
@@ -130,36 +133,76 @@ ColdLaunch.prototype.captureEntryMetadata = function(entry) {
  */
 ColdLaunch.prototype.launch = function() {
   var phase = this;
-  var marionette = phase.device.marionette;
-
-  var selector = '#icons > .icon[data-identifier*="' + phase.options.app + '"]';
-
-  if (phase.options.entryPoint) {
-    selector += '[data-identifier*="' + phase.options.entryPoint + '"]';
-  }
 
-  if (phase.options.launchDelay) {
+  if (this.options.launchDelay) {
     debug('Pausing before launching application');
   }
 
+  this.resetInput();
+
   // Delay launch to give time for pre-allocated process and system cool-down
   setTimeout(function() {
-    marionette
-      .startSession()
-      .then(function(client) {
-        marionette.switchToHomescreen();
+    phase.device.input.tap(phase.appX, phase.appY, 1);
+  }, this.options.launchDelay);
+
+  return this.waitForPerformanceEntry('fullyLoaded', this.options.app);
+};
+
+/**
+ * Cache coordinates to the target application
+ * @returns {Promise}
+ */
+ColdLaunch.prototype.getAppCoordinates = function() {
+  var phase = this;
+  var marionette = phase.device.marionette;
+  var selector = '#icons > .icon[data-identifier*="' + this.options.app + '"]';
+
+  if (this.options.entryPoint) {
+    selector += '[data-identifier*="' + this.options.entryPoint + '"]';
```

Comment:
This doesn't work, if I specify --app communcations --entrypoint contacts, it always launches the dialer app instead; AND the test actually passes anyway because it receives all the marks for "communications.gaiamobile.org"